util/src/main/java/io/zeebe/util/exception/NonRecoverableException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.util.exception;
+
+/**
+ * A non recoverable exception should wrap any exception, where it makes no sense to apply any retry
+ * strategy.
+ */
+public class NonRecoverableException extends RuntimeException {
```

Comment:
Aren't all exceptions which are not `RecoverableException` non-recoverable exceptions? What is the advantage of an explicit type here? If we have both `NonRecoverableException` and `RecoverableException`, what does that make exception types that don't derive from either?

Reply:
Yes was also not sure about that specific type, we can also simply remove it